core/src/test/java/org/apache/iceberg/rest/TestRESTCatalog.java

```diff
@@ -150,7 +154,8 @@ public <T extends RESTResponse> T execute(
     servletContext.setVirtualHosts(null);
     servletContext.setGzipHandler(new GzipHandler());
 
-    this.httpServer = new Server(8181);
+    initializePort();
+    this.httpServer = new Server(port);
```

Comment:
[doubt] Thinking of a case like, we find the free port first and then try to create a server, can it happen that in before creating a server the free port we found, can get occupied, by some other process ?

Reply:
I think the chances of that happening are pretty low as the binding port is determined by the kernel randomly from available ports. It can happen but very low as all this is happening is nano seconds at the kernel to pick a socket,  close it and then start the rest server at that port.

I think this is better than having it hard coded to 8181 as we know our CI build fails due to parallel runs